src/runtime/rpc/network.cpp

```diff
@@ -400,7 +387,7 @@ bool rpc_session::on_disconnected(bool is_write)
 
 bool rpc_session::is_auth_success(message_ex *msg)
 {
-    if (security::FLAGS_enable_auth && !_negotiation->negotiation_succeed()) {
+    if (security::FLAGS_enable_auth && !is_negotiation_succeed()) {
```

Comment:
From my point, if you want to decouple the negotiation entirely with `rpc_session`, you should remove this function and modify `rpc_session::on_recv_message`, where a session rejects the incoming requests when the negotiation is unfinished.

Nevertheless, there's no joint point for recv_message as far as I see. (There's one joint point at `task_spec::on_rpc_call`, but not suitable since it targets at customization of tasks, rather than general rpc control).

Maybe you can add such joint point later.

Reply:
Yes, I will do it in the next pull request. I split the refactor of network into 3 pull requests.